### PR TITLE
Gracefully quit on Ctrl+C

### LINKED ISSUE #268: Handle SIGTERM and SIGINT
We suggest using Ctrl+C to shutdown the process, but we never actually handle it to do any cleanup.

We should be disconnecting all the connected IRC networks and client sockets.

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #268

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -115,6 +115,31 @@ module.exports = function() {
...
+
+		// Handle ctrl+c and kill gracefully
+		let suicideTimeout = null;
+		const exitGracefully = function() {
+			if (suicideTimeout !== null) {
+				return;
+			}
+
+			// Forcefully exit after 3 seconds
+			suicideTimeout = setTimeout(() => process.exit(1), 3000);
```
Comment (astorije) [inline src/server.js]: Maybe another `log.info` in there, like "Something went wrong, force-quitting now"? Also, could do `5000` just in case, but whatever.

**Comment by astorije** [comment]
@xPaw feel free to address comment or not, then to merge or let @YaManicKill review first. I'm good either way.

File: src/server.js (hunk 1 above)
Comment (xPaw) [reply src/server.js]: I don't think it's necessary to add a message.

File: src/client.js [hunk 2]
```diff
@@ -499,7 +499,10 @@ Client.prototype.quit = function() {
 			const socket = sockets.connected[user];
 
 			if (socket) {
-				socket.emit("sign-out");
+				if (signOut) {
```
Comment (YaManicKill) [inline src/client.js]: This feels like an unrelated fix.

Comment (xPaw) [reply src/client.js]: It's not, we don't want to sign out users on server close.

Comment (YaManicKill) [reply src/client.js]: Oh woops, I missed that we were already sending sign-out. Sounds good.
